### ISSUE TITLE
Accessibility Issue On IOS - Doubling tapping a Pressable active the child button

### ISSUE THREAD

**Issue by tariqule** [facebook/react-native#32625]
### Description

I am working on a Accordion component, which has a Pressable Wrapper (which double tapping opens the accordion). There is an Icon button component (which is touchable) which is wrapped by the Pressable component. Ideally, when the accessibility mode (Voice Over on IOS) is ON then double tapping the Accordion should open the accordion rather than activating the icon button component. Currently, the icon button is getting activated even tho it was not touched. (Note: On Android its working as expected, the issue is only on IOS.)

I tried putting styles for the icon button as absolute position, then realized if the icon button is on top of the accordion still icon button is getting activated but if push icon button outside the accordion (pressable component) then i am able to press the icon and the accordion.

This is how my accordion with button looks like.

![IMG_5824 2](https://user-images.githubusercontent.com/45430120/142562032-5324d50b-39a0-498c-806d-da73dfddbcbd.PNG)

### Version

0.65.1

### Output of `react-native info`

System:
    OS: macOS 11.6
    CPU: (8) x64 Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
    Memory: 372.89 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.17.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.13 - /usr/local/bin/npm
    Watchman: 2021.06.07.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.4, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    Android SDK:
      API Levels: 23, 28, 29, 30
      Build Tools: 28.0.3, 29.0.2, 30.0.2, 30.0.3
      System Images: android-29 | Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom, android-30 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.2 AI-202.7660.26.42.7351085
    Xcode: 12.4/12D4e - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_292 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.65.1 => 0.65.1 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

- make sure the **voice over** on IOS is ON.
- create a Pressable component (or Touchable Component or any component that has onPress)
- inside the first Pressable add another Pressable component (or Touchable Component or any component that has onPress) as a child. The second pressable component can have smaller width and height so it doesn’t  overlap with the first.

Results: 
Actual: The second Pressable composite gets triggered rather than first Pressable component even though the second Pressable component was not touched.

Expected: The first Pressable component should be triggered as that pressed and double tapping on the second Pressable should trigger the second Pressable component

### Snack, code example, screenshot, or link to a repository

Example to Recreate the issue:

```
  <View>
        <Pressable
          style={{
            justifyContent: 'center',
            alignItems: 'center',
          }}
          onPress={() => {
            Alert.alert('touched outside botton')
          }}>
          <Pressable
            style={{
              width: 50,
            }}
            onPress={() => {
              Alert.alert('touched inside botton')
            }}>
            <Text>button</Text>
          </Pressable>
        </Pressable>
      </View>
```

**Comment by tariqule** [comment]
@facebook-github-bot label Accessibility

**Comment by blavalla** [comment]
So you've found an interesting edge case here. 

Generally speaking, having clickable elements nested within one another isn't a best practice (and iOS doesn't even support having nested accessible elements). But that being said let me explain what is going on here.

When VoiceOver focuses an element, and a user double-taps, it doesn't simply trigger the onClick for that element, it instead sends a synthetic touch event to the screen. This is likely so that apps that aren't written with accessibility in mind will still work properly, as there is no difference between a users finger tapping the element and this synthetic touch event.  

But since this touch event is pretending to be a users finger iOS has to decide *where* a user "touched" the screen. Since it knows the elements bounds, the logical place for it to touch is the direct center of the element, as this is the most likely place that this touch event wouldn't be interpreted to be touching something else. But in your case the exact center point of the element is where your nested button is! If you moved that button 50px to the left or right, or happened to have longer text that pushed it over automatically, this wouldn't occur.

I think your best solution here would actually be to move the info icon outside the accordion entirely, as this removes all ambiguity around what should happen when a user taps it (either with their finger, or via an accessibility service). But if you need to keep it inside the accordion, aligning it to the left or right should resolve the issue here.

As a side note, this exact same issue will happen on Android versions below 7.1, as Android had the same synthetic touch event behavior. In later versions Android defaults to triggering the elements onClick instead, and only falls back to the synthetic touch event if it sees that the onClick didn't trigger any behavior.
